### ISSUE TITLE
FetchQueue should inherit Array instead

### ISSUE THREAD

**Issue by picnicbear** [cgiffard/node-simplecrawler#318]
You seem to be modifying the Array prototype in a way that causes problems when iterating using for-in loops. As of updating to 1.0.0, all arrays in my code have the various functions of FetchQueue as properties.

```javascript 
var array = ['one', 'two', 'three'];
for(var index in array) {
  console.log(index.toString());// prints 0 1 2 add exists get (etc.)
}
```

Changing `FetchQueue.prototype = Array.prototype` to (requiring util and) `util.inherits(FetchQueue, Array)` fixes it.

**Comment by cgiffard** [comment]
Good catch — that appears to have been introduced in 864aaab28fdeb9de665dbba80698065e294ed410. I didn't notice it — apologies for that. I'll get it resolved ASAP.

**Comment by cgiffard** [comment]
This should be fixed as of dd23dc665b8b72b30431cf83ed8b396e02a708d7. Let me know if you have any issues!

**Comment by fredrikekelund** [comment]
My bad, thanks for the report @picnicbear!

**Comment by cgiffard** [comment]
For the record, this is released and available as 1.0.1 on npm. :)

**Comment by picnicbear** [comment]
No problem. Nice work on the crawler, very handy, thanks ^_^
